src/core/types/addonTypes.js

```diff
@@ -0,0 +1,82 @@
+/* @flow */
+import {
+  ADDON_TYPE_DICT,
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_LANG,
+  ADDON_TYPE_SEARCH,
+  ADDON_TYPE_THEME,
+} from 'core/constants';
+
+export type AddonVersionType = {|
+  id: number,
+  channel: string,
+  edit_url: string,
+  files: Array<Object>,
+  // The `text` property is omitted from addon.current_version.license.
+  license: { name: string, url: string },
+  reviewed: Date,
+  version: string,
+|};
+
+export type AddonAuthorType = {|
+  name: string,
+  url: string,
+|};
+
+export type AddonTypeProp =
+  typeof ADDON_TYPE_DICT |
+  typeof ADDON_TYPE_LANG |
+  typeof ADDON_TYPE_SEARCH |
+  typeof ADDON_TYPE_THEME |
+  typeof ADDON_TYPE_EXTENSION;
+
+export type AddonType = {|
+  id: number,
+  authors: Array<AddonAuthorType>,
+  average_daily_users: number,
+  categories: Object,
+  compatibility: Object,
+  current_version: AddonVersionType,
+  default_locale: string,
+  description: string,
+  edit_url: string,
+  guid: string,
+  has_eula: boolean,
+  has_privacy_policy: boolean,
+  homepage: string,
+  icon_url: string,
+  is_disabled: boolean,
+  is_experimental: boolean,
+  is_source_public: boolean,
+  name: string,
+  last_updated: Date,
+  latest_unlisted_version: ?AddonVersionType,
+  previews: Array<Object>,
+  public_stats: boolean,
+  ratings: {|
+    count: number,
+    average: number,
+  |},
+  review_url: string,
+  slug: string,
+  status: 'beta'
+    | 'lite'
```

Comment:
Oh, I see single pipes being used here too. I know the pipes can surround the object syntax and it's something special in Flow. But are they also used as regular `OR` in Flow? What would a double pipe here do? Normally I'm used to catching bitwise `OR` operators as typos so they set off alarm bells in my head. 😖